MyFirstSlideshow/app/src/main/java/com/charles/myfirstslideshow/MainActivity.java

```diff
@@ -10,12 +26,141 @@
+
+        if (savedInstanceState == null) {
+            dots1 = new View[4];
+            dots1[0] = getWindow().findViewById(R.id.dots1dot1);
+            dots1[1] = getWindow().findViewById(R.id.dots1dot2);
+            dots1[2] = getWindow().findViewById(R.id.dots1dot3);
+            dots1[3] = getWindow().findViewById(R.id.dots1dot4);
+            dots2 = new View[4];
+            dots2[0] = getWindow().findViewById(R.id.dots2dot1);
+            dots2[1] = getWindow().findViewById(R.id.dots2dot2);
+            dots2[2] = getWindow().findViewById(R.id.dots2dot3);
+            dots2[3] = getWindow().findViewById(R.id.dots2dot4);
+        }
+
+        super.onCreate(savedInstanceState);
+
+        image1 = new WeakReference<ImageView>((ImageView) getWindow().findViewById(R.id.image1));
+        image2 = new WeakReference<ImageView>((ImageView) getWindow().findViewById(R.id.image2));
+    }
+
+    public synchronized void onButtonOneClicked(View button) {
+        String currentImage = "" + images.get(i % 4);
+        i += 1;
+        // the last url of the array changes every ~ 15 minutes
+        Bitmap b;
+        try {
+            URL url = new URL(currentImage);
+            URLConnection c = url.openConnection();
+            b = BitmapFactory.decodeStream(c.getInputStream());
+        } catch (Exception e) {
+            e.printStackTrace();
+            throw new RuntimeException("Error decoding picture");
+        }
+
+        image1.get().setImageBitmap(b);
+
+        //Update dots
+        if (i % 4 == 0) {
+            dots1[0].setBackgroundColor(GREEN);
+            dots1[1].setBackgroundColor(GRAY);
+            dots1[2].setBackgroundColor(GRAY);
+            dots1[3].setBackgroundColor(GRAY);
+        }
+        if (i % 4 == 1) {
+            dots1[1].setBackgroundColor(GREEN);
+            dots1[0].setBackgroundColor(GRAY);
+            dots1[2].setBackgroundColor(GRAY);
+            dots1[3].setBackgroundColor(GRAY);
+        }
+        if (i % 4 == 2) {
+            dots1[2].setBackgroundColor(GREEN);
+            dots1[1].setBackgroundColor(GRAY);
+            dots1[0].setBackgroundColor(GRAY);
+            dots1[3].setBackgroundColor(GRAY);
+        }
+        if (i % 4 == 3) {
+            dots1[3].setBackgroundColor(GREEN);
+            dots1[1].setBackgroundColor(GRAY);
+            dots1[2].setBackgroundColor(GRAY);
+            dots1[0].setBackgroundColor(GRAY);
+        }
+    }
+
+    public synchronized void onButtonTwoClicked(View button) {
+        String currentImage = "" + images.get(j % 4);
+        j += 1;
+        // the last url of the array changes every ~ 15 minutes
+        Bitmap b;
+        try {
+            URL url = new URL(currentImage);
+            URLConnection c = url.openConnection();
+            b = BitmapFactory.decodeStream(c.getInputStream());
+        } catch (Exception e) {
+            e.printStackTrace();
+            throw new RuntimeException("Error decoding picture");
```

Comment:
As I mentioned in the comment below. You should not use you own image downloading solution (use library instead). However, I think that RuntimeException(...) is too general. Use some specific type of Exception so it will be more clear what happened in case of crash